From 3b412d523a8775001993b348c27df57938c11c77 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Thu, 9 Mar 2023 15:27:28 +0800
Subject: [PATCH] [riscv][wasm-gc] Inlining of very small wasm functions into
 JS

Port commit d9d393e442c1ebb7f2813cda3749b861a8196e3e

Change-Id: Ief9b8adeabaa3dae70cae1c895ba38ec86ae25b2
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4312602
Commit-Queue: Ji Qiu <qiuji@iscas.ac.cn>
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#86312}
---
 src/codegen/riscv/assembler-riscv-inl.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/codegen/riscv/assembler-riscv-inl.h b/src/codegen/riscv/assembler-riscv-inl.h
index 6589aaca162..30e4b1d3dad 100644
--- a/src/codegen/riscv/assembler-riscv-inl.h
+++ b/src/codegen/riscv/assembler-riscv-inl.h
@@ -66,7 +66,7 @@ void RelocInfo::apply(intptr_t delta) {
 
 Address RelocInfo::target_address() {
   DCHECK(IsCodeTargetMode(rmode_) || IsWasmCall(rmode_) ||
-         IsNearBuiltinEntry(rmode_));
+         IsNearBuiltinEntry(rmode_) || IsWasmStubCall(rmode_));
   return Assembler::target_address_at(pc_, constant_pool_);
 }
 
-- 
2.35.1

